File: src/commands/ClearCommand.java
package commands;

import commands.core.ServerSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;
import organization.OrganizationManager;

public class ClearCommand extends ServerSideCommand {

    public ClearCommand(CallbackFunction callbackFunction, OrganizationManager organizationManager) {
        super(callbackFunction, organizationManager);
    }

    @Override
    protected void executeImplementation(String[] args, CallbackFunction callback) {
        organizationManager.clear();
        callback.invoke(ExecutionStatus.SUCCESS, null);
    }
}

File: src/commands/client_side/ClearCommand.kt
package commands.client_side

import commands.client_side.core.ServerSideCommand
import database.OrganizationManagerInterface

class ClearCommand(
    organizationDatabase: OrganizationManagerInterface
) : ServerSideCommand(organizationDatabase) {
    override fun executeImplementation(argument: Any?): Result<Unit?> {
        organizationDatabase.clear()
        return Result.success(null)
    }
}
